pennylane/hf/matrices.py

```diff
@@ -87,19 +89,17 @@ def overlap(*args):
         """
         n = len(basis_functions)
         overlap_matrix = anp.eye(len(basis_functions))
-        for i, a in enumerate(basis_functions):
-            for j, b in enumerate(basis_functions):
-                if i < j:
-                    if args:
-                        args_ab = []
-                        for arg in args:
-                            args_ab.append(arg[[i, j]])
-                        overlap_integral = generate_overlap(a, b)(*args_ab)
-                    else:
-                        overlap_integral = generate_overlap(a, b)()
-                    o = anp.zeros((n, n))
-                    o[i, j] = o[j, i] = 1.0
-                    overlap_matrix = overlap_matrix + overlap_integral * o
+
+        for (i, a), (j, b) in it.combinations(enumerate(basis_functions), r=2):
```

Comment:
Just double checking: it contains the condition `if i < j`? If this condition is not there, the tests will still pass but we will be computing repetitive terms. So the efficiency will be affected.